turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -272,30 +273,39 @@ impl DiskFileSystem {
 
     pub fn invalidate(&self) {
         let _span = tracing::info_span!("invalidate filesystem", path = &*self.root).entered();
-        for (_, invalidators) in take(&mut *self.invalidator_map.lock().unwrap()).into_iter() {
-            invalidators.into_iter().for_each(|i| i.invalidate());
-        }
-        for (_, invalidators) in take(&mut *self.dir_invalidator_map.lock().unwrap()).into_iter() {
-            invalidators.into_iter().for_each(|i| i.invalidate());
-        }
+        let span = tracing::Span::current();
+        let invalidator_map = take(&mut *self.invalidator_map.lock().unwrap());
+        let dir_invalidator_map = take(&mut *self.dir_invalidator_map.lock().unwrap());
+        let iter = invalidator_map
+            .into_par_iter()
+            .chain(dir_invalidator_map.into_par_iter())
+            .flat_map(|(_, invalidators)| invalidators.into_par_iter());
+        iter.for_each(|i| {
```

Comment:
IIRC, rayon iterators are designed for cpu-heavy tasks and it does not perform well for mutex-related things

Reply:
A very recent addition to this space is https://github.com/dragostis/chili

Or we could just call `spawn_blocking` a bunch of times (once per cpu core?) and re-use tokio's blocking thread pool.

I don't love pulling in a heavy dependency like rayon, but it looks like we're already pulling it in for other reasons so 🤷

Reply:
Yeah, I also dislike using rayon here, but it's cpu-heavy work here and tokio doesn't offer parallel iterators. We could rebuild that with spawn_blocking, but that sounds like it will be less efficient compared to rayon...